src/ssr-template-helper.js

```diff
@@ -113,34 +107,36 @@ export class SsrTemplateHelper {
    */
   buildPayload_(
     request, mustacheTemplate, opt_templates, opt_attributes = {}) {
-    const ampComponent = dict({
-      'type': this.sourceComponent_,
-      'successTemplate': {
+    const ampComponent = dict({'type': this.sourceComponent_});
+    if ((opt_templates && opt_templates['successTemplate'])
+        || mustacheTemplate) {
```

Comment:
How about we save the result of this expression into a `successTemplate` variable so that you don't have to reevaluate it later? You can just do `'payload': successTemplate.innerHTML` in the object literal.

Reply:
Done.